core/templates/dev/head/pages/exploration-editor-page/improvements-tab/services/improvement-modal.service.ts

```diff
@@ -21,22 +21,100 @@ require(
+            'answer-details-modal.template.html'),
+          resolve: {
+            isUserLoggedIn: function() {
+              return UserService.getUserInfoAsync().then(function(userInfo) {
+                return userInfo.isLoggedIn();
+              });
+            },
+          },
+          controller: [
+            '$scope', '$uibModalInstance', 'isUserLoggedIn',
+            function($scope, $uibModalInstance, isUserLoggedIn) {
+              $scope.selectedLearnerAnswerInfo = [];
+              $scope.learnerAnswerDetails = learnerAnswerDetails;
+              $scope.currentLearnerAnswerInfo = null;
+              $scope.viewAnswerDetails = false;
+              $scope.changeView = function(learnerAnswerInfo) {
+                $scope.currentLearnerAnswerInfo = learnerAnswerInfo;
+                $scope.viewAnswerDetails = !($scope.viewAnswerDetails);
+              };
+              $scope.getLocaleAbbreviatedDatetimeString = (
+                DateTimeFormatService.getLocaleAbbreviatedDatetimeString);
+              $scope.getLearnerAnswerInfos = function() {
+                return $scope.learnerAnswerDetails.learnerAnswerInfoData;
+              };
+              $scope.getAnswerDetails = function(learnerAnswerInfo) {
+                return learnerAnswerInfo._answerDetails;
+              };
+              $scope.selectLearnerAnswerInfo = function(learnerAnswerInfo) {
+                var index = $scope.selectedLearnerAnswerInfo.indexOf(
+                  learnerAnswerInfo);
+                if (index === -1) {
+                  $scope.selectedLearnerAnswerInfo.push(learnerAnswerInfo);
+                } else {
+                  $scope.selectedLearnerAnswerInfo.splice(index, 1);
+                }
+              };
+              $scope.deleteSelectedLearnerAnswerInfo = function() {
+                for (
+                  var i = 0; i < $scope.selectedLearnerAnswerInfo.length; i++) {
+                  var index = (
+                    $scope.learnerAnswerDetails.learnerAnswerInfoData.indexOf(
+                      $scope.selectedLearnerAnswerInfo[i]));
+                  $scope.learnerAnswerDetails.learnerAnswerInfoData.splice(
+                    index, 1);
+                  LearnerAnswerDetailsDataService.deleteLearnerAnswerInfo(
+                    $scope.learnerAnswerDetails.expId,
+                    $scope.learnerAnswerDetails.stateName,
+                    $scope.selectedLearnerAnswerInfo[i]._id);
+                }
+                $scope.selectedLearnerAnswerInfo = [];
+                $scope.currentLearnerAnswerInfo = null;
+                if ($scope.getLearnerAnswerInfos().length === 0) {
+                  $scope.close();
+                }
+              };
+              $scope.getLearnerAnswerHtml = function(learnerAnswerInfo) {
+                return ExplorationHtmlFormatterService.getShortAnswerHtml(
+                  learnerAnswerInfo._answer,
```

Comment:
Ditto, (`getAnswer()`) here and elsewhere where fields were directly used

Reply:
Done